File: Habitica/src/com/magicmicky/habitrpgwrapper/lib/models/Egg.java
package com.magicmicky.habitrpgwrapper.lib.models;

import com.habitrpg.android.habitica.HabitDatabase;
import com.raizlabs.android.dbflow.annotation.Column;
import com.raizlabs.android.dbflow.annotation.Table;

@Table(databaseName = HabitDatabase.NAME)
public class Egg extends BaseItem {

    @Column
    String adjective, mountText;

    public String getAdjective() {
        return adjective;
    }

    public void setAdjective(String adjective) {
        this.adjective = adjective;
    }

    public String getMountText() {
        return mountText;
    }

    public void setMountText(String mountText) {
        this.mountText = mountText;
    }
}

File: Habitica/src/main/java/com/habitrpg/android/habitica/models/inventory/Egg.kt
package com.habitrpg.android.habitica.models.inventory

import io.realm.RealmObject
import io.realm.annotations.PrimaryKey

open class Egg : RealmObject(), Item {
    @PrimaryKey
    override var key: String = ""
    override var text: String = ""
    var notes: String? = null
    override var value: Int = 0
    var adjective: String? = null
    var mountText: String? = null

    override val type: String
        get() = "eggs"
}